modules/wallet/seed_test.go

```diff
@@ -278,57 +278,71 @@ func TestSweepSeedFunds(t *testing.T) {
 	// wallet. This wallet should have a siafund balance.
 	//
 	// TODO: when proper seed loading is implemented, this will be unnecessary.
-	w, err := New(wt.cs, wt.tpool, wt.wallet.persistDir)
+	wt.wallet, err = New(wt.cs, wt.tpool, wt.wallet.persistDir)
 	if err != nil {
 		t.Fatal(err)
 	}
-	defer w.Close()
 	// reset the changeID
-	resetChangeID(w)
-	err = w.Unlock(wt.walletMasterKey)
+	resetChangeID(wt.wallet)
+	err = wt.wallet.Unlock(wt.walletMasterKey)
 	if err != nil {
 		t.Fatal(err)
 	}
-	_, siafundBal, _ := w.ConfirmedBalance()
+	_, siafundBal, _ := wt.wallet.ConfirmedBalance()
 	if siafundBal.Cmp(types.NewCurrency64(2000)) != 0 {
 		t.Error("expecting a siafund balance of 2000 from the 1of1 key")
 	}
+	// need to reset the miner as well, since it depends on the wallet
+	wt.miner, err = miner.New(wt.cs, wt.tpool, wt.wallet, wt.wallet.persistDir)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// send funds to ourself a few times
+	for i := 0; i < 10; i++ {
+		uc, err := wt.wallet.NextAddress()
+		if err != nil {
+			t.Fatal(err)
+		}
+		_, err = wt.wallet.SendSiafunds(types.NewCurrency64(1), uc.UnlockHash())
+		if err != nil {
+			t.Fatal(err)
+		}
+		wt.addBlockNoPayout()
+	}
+	// send some funds to the void
+	_, err = wt.wallet.SendSiafunds(types.NewCurrency64(10), types.UnlockHash{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	wt.addBlockNoPayout()
```

Comment:
That is what I mean.

DRY is good in testing but only if you aren't too creative about it. If you end up changing the setup code, you risk breaking any test that's depending on it. But if it's all copy-paste, that risk isn't there. It also means having to learn more helper functions which might be spread across multiple files and might have their own helper functions (api package testing is like this), and that can get annoying / confusing / frustrating if you are just trying to understand / fix a single test.